tests/prefetch_related/tests.py

```diff
@@ -906,6 +907,24 @@ def test_m2m(self):
             )
 
 
+class ShadowTest(TestCase):
+    @ignore_warnings(category=RemovedInDjango20Warning)
```

Comment:
Are you sure this code path is still triggered now that you raise a `ValueError`?